backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/sleketon/BpkSkeleton.kt

```diff
@@ -0,0 +1,186 @@
+import net.skyscanner.backpack.compose.tokens.BpkColor
+import net.skyscanner.backpack.compose.utils.dynamicColorOf
+
+private val BpkImageSkeletonSize = 96.dp
+private val BpkCircleSkeletonSizeSm = 32.dp
+private val BpkCircleSkeletonSizeLg = 48.dp
+private val BpkSkeletonBorderRadiusXXS = 2.dp
+private val BpkHeadlineSkeletonWidth = 80.dp
+private val BpkHeadlineSkeletonHeightSm = 8.dp
+private val BpkHeadlineSkeletonHeightMd = 16.dp
+private val BpkHeadlineSkeletonHeightLg = 32.dp
+
+enum class BpkSkeletonSizeType {
+    Small,
+    Medium,
+    Large
+}
+
+enum class BpkCircleSkeletonSizeType {
+    Small,
+    Large
+}
+
+enum class BpkSkeletonCornerType {
+    Square,
+    Rounded
+}
+
+@Composable
+private fun shimmerBackgroundColor() : Color {
+    return dynamicColorOf(BpkColor.SkyGrayTint06, BpkColor.BlackTint02)
+}
+
+@Composable
+private fun shimmerPrimaryColor() : Color {
+    return dynamicColorOf(Color(0x00FFFFFF), Color(0x00000000))
+}
+
+@Composable
+private fun shimmerSecondaryColor() : Color {
+    return dynamicColorOf(Color(0x99FFFFFF), Color(0x33000000))
+}
+
+@Composable
+private fun shimmerAnimation(): State<Dp> {
+    var inifiniteTransition = rememberInfiniteTransition()
+    return inifiniteTransition.animateValue(initialValue = (-500).dp, targetValue = 500.dp, typeConverter = Dp.VectorConverter,
+        animationSpec = infiniteRepeatable(animation = tween(durationMillis = 1000, delayMillis = 200, easing = LinearEasing)))
+
+}
+@Composable
+private fun ShimmerBox () {
+    return Box(modifier = Modifier
+        .size(1000.dp, 1000.dp)
+        .background(
+            brush = Brush.horizontalGradient(
+                listOf(
+                    shimmerPrimaryColor(),
+                    shimmerSecondaryColor(),
+                    shimmerPrimaryColor(),
+                )
+            )
+        )
+    )
+}
+
+@Preview
+@Composable
+fun BpkImageSkeleton (
+    modifier: Modifier = Modifier,
+    cornerType: BpkSkeletonCornerType = BpkSkeletonCornerType.Square
+) {
+    val offsetX by shimmerAnimation()
+    Box(modifier = modifier
+        .size(BpkImageSkeletonSize, BpkImageSkeletonSize)
+        .background(shimmerBackgroundColor(),
+            RoundedCornerShape(if (cornerType === BpkSkeletonCornerType.Rounded) BpkBorderRadius.Sm else 0.dp))
+        .offset(Dp(offsetX.value), 0.dp)) {
+        ShimmerBox()
+    }
+}
+
+@Preview
+@Composable
+fun BpkBodyTextSkeleton (
+    modifier: Modifier = Modifier,
+    skeletonSize:  BpkSkeletonSizeType= BpkSkeletonSizeType.Small
+) {
+    val offsetX by shimmerAnimation()
+
+    val skeletonWidthPercentage = when (skeletonSize) {
+        BpkSkeletonSizeType.Small -> 0.42F
+        BpkSkeletonSizeType.Medium -> 0.62F
+        BpkSkeletonSizeType.Large -> 0.73F
+    }
+
+    Box(modifier = modifier){
+        Box(modifier = modifier
+            .fillMaxWidth(skeletonWidthPercentage)
+            .height(8.dp)
```

Comment:
could you use the `Md` spacing token for this?

Reply:
ok.